builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -642,3 +663,26 @@ func dropGeneratedSecurityGroup(settingValue string, meta interface{}) string {
 
 	return strings.Join(legitGroups, ",")
 }
+
+func describeBeanstalkEvents(conn *elasticbeanstalk.ElasticBeanstalk, environmentId string, t time.Time) error {
```

Comment:
A better name might be `describeBeanstalkErrors` since this only checks for ERRORs.

I'd also prefer a `[]string` of the errors, (or even a `[]*elasticbeanstalk.EventDescription` rather than concatenating them into an error value. A function returning an error generally indicated the call failed.